File: Insertion Sort/InsertionSort.py
def insertionSort(a):
    for j in range(1, len(a)):
        key = a[j]
        i = j-1
        while i>=0 and a[i] > key:
            a[i+1] = a[i]
            print(a)
            i-=1
        a[i+1] = key 
        print(a)

array = [6, 5, 4, 3, 2, 1]
insertionSort(array)

File: Insertion Sort/InsertionSort.ts
import misc = require("../_Miscellaneous/misc");

export function insertionSort<T>(a: T[]): void{
    for(let j = 1; j<a.length; j++){
        let key = a[j];
        let i = j-1;
        while(i>=0 && a[i] > key){
            a[i+1] = a[i]
            i--;
        }
        a[i+1] = key;
    }
}